hoodie-common/src/main/java/com/uber/hoodie/common/util/SpillableMapUtils.java

```diff
@@ -0,0 +1,155 @@
+import java.io.RandomAccessFile;
+import java.util.Optional;
+import java.util.zip.CRC32;
+
+public class SpillableMapUtils {
+
+  /**
+   * Using the schema and payload class, read value from disk and return a HoodieRecord
+   * @param file
+   * @param schema
+   * @param payloadClazz
+   * @param valuePosition
+   * @param valueLength
+   * @param <R>
+   * @return
+   * @throws IOException
+   */
+  public static <R> R read(RandomAccessFile file, Schema schema, String payloadClazz,
+                           long valuePosition, int valueLength) throws IOException {
+
+    DiskBasedMap.FileEntry fileEntry = readInternal(file, valuePosition, valueLength);
+    return (R) convertBytesToRecord(fileEntry.getValue(), schema, payloadClazz);
+  }
+
+  /**
+   * |crc|timestamp|sizeOfKey|SizeOfValue|key|value|
+   * @param file
+   * @param valuePosition
+   * @param valueLength
+   * @return
+   * @throws IOException
+   */
+  private static DiskBasedMap.FileEntry readInternal(RandomAccessFile file, long valuePosition, int valueLength) throws IOException {
+    file.seek(valuePosition);
+    long crc = file.readLong();
+    long timestamp = file.readLong();
+    int keySize = file.readInt();
+    int valueSize = file.readInt();
+    byte [] key = new byte[keySize];
+    file.read(key, 0, keySize);
+    byte [] value = new byte[valueSize];
+    if(!(valueSize == valueLength)) {
+      throw new HoodieCorruptedDataException("unequal size of payload written to external file");
+    }
+    file.read(value, 0, valueSize);
+    long crcOfReadValue = SpillableMapUtils.generateChecksum(value);
+    if(!(crc == crcOfReadValue)) {
+      throw new HoodieCorruptedDataException("checksum of payload written to external disk does not match");
+    }
+    return new DiskBasedMap.FileEntry(crc, keySize, valueSize, key, value, timestamp);
+  }
+
+  /**
+   * Write Value and other metadata necessary to disk. Each entry looks as follows:
+   *
+   * |crc|timestamp|sizeOfKey|SizeOfValue|key|value|
+   *
+   * @param outputStream
+   * @param fileEntry
+   * @return
+   * @throws IOException
+   */
+  public static long write(SizeAwareDataOutputStream outputStream, DiskBasedMap.FileEntry fileEntry) throws IOException {
+    return writeInternal(outputStream, fileEntry);
+  }
+
+
+  private static long writeInternal(SizeAwareDataOutputStream outputStream, DiskBasedMap.FileEntry fileEntry)
+      throws IOException {
+    outputStream.writeLong(fileEntry.getCrc());
+    outputStream.writeLong(fileEntry.getTimestamp());
+    outputStream.writeInt(fileEntry.getSizeOfKey());
+    outputStream.writeInt(fileEntry.getSizeOfValue());
+    outputStream.write(fileEntry.getKey());
+    outputStream.write(fileEntry.getValue());
+    outputStream.flush();
+    return outputStream.getSize();
+  }
+
+  /**
+   * Generate a checksum for a given set of bytes
+   * @param data
+   * @return
+   */
+  public static long generateChecksum(byte [] data) {
+    CRC32 crc = new CRC32();
```

Comment:
How expensive is this constructor?  alternative is to reset this everytime and then update() as needed

Reply:
/**
     * Creates a new CRC32 object.
     */
    public CRC32() {
    }
That's all it does.